pyomo/core/base/block.py

```diff
@@ -702,15 +707,56 @@ def __delattr__(self, name):
             #
             super(_BlockData, self).__delattr__(name)
 
-    def set_value(self, val):
-        for k in list(getattr(self, '_decl', {})):
-            self.del_component(k)
-        self._ctypes = {}
-        self._decl = {}
-        self._decl_order = []
-        if val:
-            for k in sorted(iterkeys(val)):
-                self.add_component(k,val[k])
+    def set_value(self, val, guarantee_components=()):
```

Comment:
You might want to clarify `guarantee_components`

Reply:
I agree with @qtothec, I don't really understand what `guarantee_components` is doing